src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/SelfRefreshingKeySet.java

```diff
@@ -119,11 +119,11 @@ private JsonWebKey getKeyWithKeyId(String kid) throws AuthenticatorUnavailableEx
         return result;
     }
 
-    private synchronized JsonWebKey getKeyWithRefresh(String kid) throws AuthenticatorUnavailableException {
+    private synchronized JWK getKeyWithRefresh(String kid) throws AuthenticatorUnavailableException {
```

Comment:
why the method is `synchronized`? 
is it a global object? if it creates per request there is no need to synchronize it